elevate-cpanel

```diff
@@ -2897,61 +2907,224 @@ sub _latest_checksum () {
-        server provider to: https://github.com/cpanel/elevate/issues/101
-        EOS
-    }
+    or to change it so that it is set to "true" instead.
+    EOS
 
     return 0;
 }
 
 sub _parse_shell_variable ( $path, $varname ) {
 
-    my (undef, $dir, $file) = File::Spec->splitpath($path);
+    my ( undef, $dir, $file ) = File::Spec->splitpath($path);
     $dir = File::Spec->canonpath($dir);
 
     # drop privileges and run bash in restricted mode
     my $bash_sr = Cpanel::SafeRun::Object->new(
         program => Cpanel::Binaries::path('bash'),
-        args => [
+        args    => [
             '--restricted',
             '-c',
             qq(set -ux ; [ "x\$PWD" = "x$dir" ] || exit 72 ; source $file ; echo "\$$varname"),
         ],
         before_exec => sub {
-            chdir $dir; 
+            chdir $dir;
             Cpanel::AccessIds::SetUids::setuids('nobody');
         },
     );
 
     return undef if $bash_sr->CHILD_ERROR && $bash_sr->error_code == 127;
-    $bash_sr->die_if_error(); # bail out if something else went wrong
+    $bash_sr->die_if_error();    # bail out if something else went wrong
 
     my $value = $bash_sr->stdout;
     chomp $value;
     return $value;
 }
 
+sub GRUB2_PREFIX_DEBIAN { return '/boot/grub' }
+sub GRUB2_PREFIX_RHEL   { return '/boot/grub2' }
+
+sub _blocker_grub2_workaround ($self) {
+    my $state = _grub2_workaround_state();
+    if ( $state == GRUB2_WORKAROUND_OLD ) {
+        my ( $deb, $rhel ) = ( GRUB2_PREFIX_DEBIAN, GRUB2_PREFIX_RHEL );
+        WARN( <<~EOS );
+        $deb/grub.cfg is currently a symlink to $rhel/grub.cfg. Your provider
+        may have added this to support booting your server using their own
+        instance of the GRUB2 bootloader, one which looks for its
+        configuration, boot entries, and modules in a different location from
+        where your operating system stores this data. In order to allow the
+        process to complete successfully, the upgrade process will rename the
+        current $deb directory, re-create $deb as a symlink to $rhel,
+        and then copy as much of the old $deb into $rhel as possible.
+        EOS
+        update_stage_file( { 'grub2_workaround' => { 'needs_workaround_update' => 1 } } ) if $self->{_abort_on_first_blocker};    # don't update stage file if this is just a check
+    }
+    elsif ( $state == GRUB2_WORKAROUND_UNCERTAIN ) {
+
+        return $self->has_blocker( 107, <<~EOS );
+        The configuration of the GRUB2 bootloader does not match the
+        expectations of this script. For more information, see the output of
+        the script when run at the console:
+
+        /scripts/elevate-cpanel --check
+
+        If your GRUB2 configuration has not been customized, consider reporting
+        this limitation to https://github.com/cpanel/elevate/issues
+        EOS
+    }
+
+    return 0;
+}
+
+# TODO: This should probably return warnings/errors rather than directly printing to the log.
```

Comment:
Outstanding TODO?

Reply:
It's more "something that could be nicer to work with" TODO than "thing I absolutely need to implement" TODO.